### PR TITLE
Added precommit hook

### LINKED ISSUE #62: Add precommit Hook for code formatting
The code should be automatically formatted while a commit is made. 

You may use this [resource](https://www.freecodecamp.org/news/how-to-add-commit-hooks-to-git-with-husky-to-automate-code-tasks/) for reference. 

There's a prettier config file in this repository. Use it to format code before commit. You can use [Husky and/or pretty-quick](https://prettier.io/docs/en/precommit.html#option-2-pretty-quickhttpsgithubcomazzpretty-quick) (if you want)

### PR DESCRIPTION
**Pull request by Samridhi-98**
# Description

Added pre-commit hook  `.pre-commit-config.yaml`

Fixes #62

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@Manvityagi  Please review my PR.

**Comment by abdus** [comment]
hey @Samridhi-98, did you test these changes locally before making a Pull Request? from what I can see, you have installed `husky@6` for this project. But the way you have implemented it .. works with `husky@4`. so, the pre-commit hooks never actually get triggered. 

following are the changes I want you to make:
1. remove `.pre-commit-config.yaml. this is required only if you are using [pre-commit](https://pre-commit.com/) framework
2. migrate the implementation to husky@6. resources: [installation](https://typicode.github.io/husky/#/?id=install)  &middot; [migration](https://typicode.github.io/husky/#/?id=migrate-from-v4-to-v6)
3. update readme with relevant information (for example, if there's any additional commands user required to run while setting up the project locally)

Let me know if you need any help. And please test your code locally before making PRs.

**Comment by Samridhi-98** [comment]
> hey @Samridhi-98, did you test these changes locally before making a Pull Request? from what I can see, you have installed `husky@6` for this project. But the way you have implemented it .. works with `husky@4`. so, the pre-commit hooks never actually get triggered.
> 
> following are the changes I want you to make:
> 
> 1. remove `.pre-commit-config.yaml. this is required only if you are using [pre-commit](https://pre-commit.com/) framework
> 2. migrate the implementation to husky@6. resources: [installation](https://typicode.github.io/husky/#/?id=install)  · [migration](https://typicode.github.io/husky/#/?id=migrate-from-v4-to-v6)
> 3. update readme with relevant information (for example, if there's any additional commands user required to run while setting up the project locally)
> 
> Let me know if you need any help. And please test your code locally before making PRs.

Okay, will update as you suggested.

**Comment by Samridhi-98** [comment]
@abdus  Please review I have updated as you suggested and also tested my code locally.

**Screenshot**

ss

File: README.md [hunk 1]
```diff
@@ -31,6 +31,9 @@ $ npm install
 # Copy .env.example to .env
 $ It is recommended to install [MongoDB Compass](https://www.mongodb.com/try/download/compass) to visualize data in mongodb
 
+# To test run pre commit hook 
+$ npm test
+
```
Comment (abdus) [inline README.md]: `pre-commit` hook should run automatically before committing the changes.

File: package.json [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
+        "swagger-jsdoc": "^6.0.0",
+        "swagger-ui-express": "^4.1.6"
+    },
+    "devDependencies": {
+        "chai": "^4.3.4",
+        "faker": "^5.5.2",
+        "husky": "^6.0.0",
+        "mocha": "^8.3.2",
+        "sinon": "^10.0.0"
+    }
```
Comment (abdus) [inline package.json]: you need to install `prettier` and `pretty-quick` as well.

**Review by abdus — CHANGES_REQUESTED** [review]
The changes you have made aren't working as expected. Files that were changed since the last commit, should be formatted automatically using prettier before commit. 

Also, by "test your code locally", I meant to say that whatever code you have changed, is working as expected. For example, here you are working on formatting code with the pre-commit hook. check that whether the changed files were formatted or not.

I wrote a [how-to on writing a pre-commit hook using husky@6](https://log.abdus.net/#Sun,%2025%20Apr%202021%2000:00:00%20GMT). please check it out and see if this could be of any help.

**Comment by Samridhi-98** [comment]
@abdus  Please review. Also while running this `npx husky add .husky/pre-commit './node_modules/.bin/pretty-quick --staged'` got an error so I have created the file using `npx husky add .husky/pre-commit` first and then added this  `'./node_modules/.bin/pretty-quick --staged'` manually and its working, please let me know if I did anything wrong.

**Screenshot**
github

**Review by abdus — APPROVED** [review]

**Comment by Samridhi-98** [comment]
@abdus @Manvityagi   please add the appropriate label and GSSOC tag.

**Pull request merged**